spec/controllers/rentals_controller_spec.rb

```diff
@@ -156,15 +156,27 @@
     end
   end
 
+  describe 'GET #invoice' do
+    it 'assigns a requested rental to @rental' do
+      get :invoice, params: { id: @rental }
+      expect(assigns[:rental]).to eq @rental
+    end
+
+    it 'all requested financial transactions should contain the same rental as @rental' do
+      get :show, params: { id: @rental }
+      expect(assigns[:financial_transactions].all? { |ft| ft.rental.id == @rental.id }).to be true
```

Comment:
Does set difference work `expect(assigns[:financial_transactions].map(:rental_id) - [@rental.id]).to be_empty`
work by any chance?
Or perhaps`expect(assigns[:financial_transactions].map(:rental_id).uniq).to eq([@rental.id])`